frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellAdminSetup.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { k8sCreate, k8sGet } from '@console/internal/module/k8s';
+import { WorkspaceModel } from '../../../models';
+import {
+  newCloudShellWorkSpace,
+  createCloudShellResourceName,
+  CLOUD_SHELL_PROTECTED_NAMESPACE,
+} from '../cloud-shell-utils';
+import { NamespaceModel } from '@console/internal/models';
+import TerminalLoadingBox from '../TerminalLoadingBox';
+import { LoadError } from '@console/internal/components/utils/status-box';
+import { useTranslation } from 'react-i18next';
+
+type Props = {
+  onInitialize?: (namespace: string) => void;
+};
+
+const CloudShellAdminSetup: React.FunctionComponent<Props> = ({ onInitialize }) => {
+  const { t } = useTranslation();
+
+  const [initError, setInitError] = React.useState<string>();
+
+  React.useEffect(() => {
+    (async () => {
+      async function namespaceExists(): Promise<boolean> {
+        try {
+          await k8sGet(NamespaceModel, CLOUD_SHELL_PROTECTED_NAMESPACE);
+          return true;
+        } catch (error) {
+          const namespaceError = `namespaces "${CLOUD_SHELL_PROTECTED_NAMESPACE}" not found`;
+          if (error.message !== namespaceError) {
```

Comment:
I don't like this error message comparison. Can you use the status code instead? eg 404